[cvlistitem] Space control between list symbol and text feature (#74)

LLdaniel opened the pull request:
#### Context
As far as I know it is in general a bit more complicated when having parameters with multiple default values. 
Maybe it could also be discussed if more default parameters distract more than they help. 

#### Ideas
Two (three) ideas for implementing the enhancement of #74:

1. This PR utilizing the `xargs` package with the disadvantage of introducing an additional package dependency
2. I could also think of defining a new `listitemsymbolspacing` which can be redefined with the help of `setlength` and therefore introduces no additional package dependency. 
3. I am also open to completely new suggestions and can try to implement those

jalopezg-git commented:
Hi @LLdaniel,

I have just added myself for review :slightly_smiling_face: here.  is this PR ready now?

LLdaniel commented:
Hi @jalopezg-git ,
yes this commit tries to solve the described issue with option 1. I would be glad if you could review it. Please let me know if this would be a option or maybe we should switch to another option. 
Thanks in advance!

jalopezg-git commented on template.tex:
```diff
@@ -218,6 +218,15 @@ \section{Interests}
 \section{Extra 1}
+%% Fine tuned examples due to two default parameters:
+%% - full specification command:
+%%   cvlistitem[vertical spacing between list items][space after list symbol]{Item}
```
```suggestion
%%   \cvlistitem[vertical spacing between list items][space after list symbol]{Item}
```

LLdaniel replied:
not needed anymore for option 2

jalopezg-git commented on moderncvbodyi.sty:
```diff
@@ -97,8 +97,8 @@
+\renewcommandx*{\cvlistitem}[3][1=.25em, usedefault, 2=0em]{%
+  \cvitem[#1]{}{\listitemsymbol\hspace{#2}\begin{minipage}[t]{\listitemcolumnwidth - #2}#3\end{minipage}}}
 
 \renewcommand*{\cvlistdoubleitem}[3][.25em]{%
```
I think that additional space between `\listitemsymbol` and the text should also be applied for `\cvlistdoubleitem{...}`.

LLdaniel replied:
new implementation with option 2

jalopezg-git reviewed (changes requested):
Thanks for working on this, @LLdaniel!

I'm not too worried about the dependency on `xargs`.  However, contrarily to the vertical-spacing (which the user might want to vary depending on the list), the symbol-to-text spacing looks more a global/style setting that should affect the overall look.

That said, I think your second proposal might make more sense for this, namely
>  2.  I could also think of defining a new `\listitemsymbolspacing` which can be redefined with the help of `\setlength` and therefore introduces no additional package dependency.

What do you think?

Also, `moderncvbodyii.sty` and `moderncvbodyvi.sty` (introduced by the 'contemporary' style) should also need updating.

LLdaniel commented:
Hi @jalopezg-git, 
thanks for your feedback! I am happy about the discussion and have to admit that your argument makes totally sense:
This should be more an overall option.
Therefore I reverted option 1 and implemented a first version of option 2 with [10cd637](https://github.com/moderncv/moderncv/pull/182/commits/10cd63728912d74623700e5dc49429e55093f520).

I hope `pt` is a good default unit for that. I choose that because of the calculation with `\textwidth`. 

At the moment I am also unsure what to do with `\cvcolumns` in the "References" section: Do you think we should try to adjust them as well for a more consistent impression? Otherwise I would leave that for the moment as it is.

jalopezg-git reviewed (commented):
Thanks, @LLdaniel!  I think that, in general, this approach makes more sense :+1:.

LGTM (except that we are missing changes in `moderncvbodyii.sty`).  Is this intentional?

> At the moment I am also unsure what to do with \cvcolumns in the "References" section: Do you think we should try to adjust them as well for a more consistent impression? Otherwise I would leave that for the moment as it is.

For now, I'll leave as-is until we have a good reason to make those changes.

LLdaniel commented:
Hi @jalopezg-git ,
thank you for your review!

Regarding the `moderncvbodii.sty`:
I left it out because it is a symlink to `moderncvbodi.sty`:
```bash
ll moderncvbodyii.sty 
lrwxrwxrwx 1 daniel daniel 17 Mar 28  2023 moderncvbodyii.sty -> moderncvbodyi.sty
```

I hope the changes should be propagated correctly then. If I am missing something, please hit me a comment and I will fix that.

jalopezg-git commented:
> Hi @jalopezg-git , thank you for your review!
> 
> Regarding the `moderncvbodii.sty`: I left it out because it is a symlink to `moderncvbodi.sty`:
> 
> ```shell
> ll moderncvbodyii.sty 
> lrwxrwxrwx 1 daniel daniel 17 Mar 28  2023 moderncvbodyii.sty -> moderncvbodyi.sty
> ```
> 
> I hope the changes should be propagated correctly then. If I am missing something, please hit me a comment and I will fix that.

Oh, sorry!  It's me who missed that :rofl: - too much things running around on my head.  LGTM then!

LLdaniel commented:
> Oh, sorry! It's me who missed that 🤣 - too much things running around on my head. LGTM then!

No worries! I am a big fan of a code review. It helps to think about the implementation twice and normally I am the one who misses something out 🤣
Thanks for your help and with this we can merge another PR 🚀

jalopezg-git commented:
@LLdaniel feel free to add me as a reviewer for the PRs that you want me to take a look on!

Have a nice day :slightly_smiling_face:!
